pkg/autoscaler/health.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package autoscaler
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/golang/glog"
+)
+
+type healthInfo struct {
+	m   *sync.RWMutex
```

Comment:
Unless we expect heavy concurrency of readers, RWMutex is overkill and cognitive burned.  Simple Mutex should suffice.

Reply:
And why is this a pointer?  It should not need to be - copying this struct by value would be wrong.

Reply:
ok right, fixing